x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go

```diff
@@ -234,10 +221,16 @@ func metricDescriptor(ctx context.Context, client *monitoring.MetricClient, proj
 				return metricsWithMeta, errors.Errorf("Could not make ListMetricDescriptors request: %s: %v", mt, err)
 			}
 
-			metricsWithMeta[mt] = metricMeta{
+			meta := metricMeta{
 				samplePeriod: time.Duration(out.Metadata.SamplePeriod.Seconds) * time.Second,
-				ingestDelay:  time.Duration(out.Metadata.IngestDelay.Seconds) * time.Second,
+				ingestDelay:  0 * time.Second,
+			}
+
+			if out.Metadata.IngestDelay != nil {
```

Comment:
I see that you check nil here now, would that be a good check on the `out.Metadata.SamplePeriod`?

Reply:
Good point, I have not seen nil for SamplePeriod but it's always safe to add a check. Thanks!